frame/two-phase-election-provider/src/benchmarking.rs

```diff
@@ -0,0 +1,358 @@
+	} verify {
+		assert!(<TwoPhase<T>>::current_phase().is_off());
+	}
+
+	on_initialize_open_signed {
+		// NOTE: this benchmark currently doesn't have any components because the length of a db
+		// read/write is not captured. Otherwise, it is quite influenced by how much data
+		// `T::ElectionDataProvider` is reading and passing on.
+		assert!(<TwoPhase<T>>::snapshot().is_none());
+		assert!(<TwoPhase<T>>::current_phase().is_off());
+	}: {
+		<TwoPhase<T>>::on_initialize_open_signed();
+	} verify {
+		assert!(<TwoPhase<T>>::snapshot().is_some());
+		assert!(<TwoPhase<T>>::current_phase().is_signed());
+	}
+
+	on_initialize_open_unsigned {
+		assert!(<TwoPhase<T>>::snapshot().is_none());
+		assert!(<TwoPhase<T>>::current_phase().is_off());
+	}: {
+		<TwoPhase<T>>::on_initialize_open_unsigned(Phase::Off, 1u32.into());
+	} verify {
+		assert!(<TwoPhase<T>>::snapshot().is_some());
+		assert!(<TwoPhase<T>>::current_phase().is_unsigned());
+	}
+
+	finalize_signed_phase_accept_solution {
+		let receiver = account("receiver", 0, SEED);
+		let initial_balance = T::Currency::minimum_balance() * 10u32.into();
+		T::Currency::make_free_balance_be(&receiver, initial_balance);
+		let ready: ReadySolution<T::AccountId> = Default::default();
+		let deposit: BalanceOf<T> = 10u32.into();
+		let reward: BalanceOf<T> = 20u32.into();
+
+		assert_ok!(T::Currency::reserve(&receiver, deposit));
+		assert_eq!(T::Currency::free_balance(&receiver), initial_balance - 10u32.into());
+	}: {
+		<TwoPhase<T>>::finalize_signed_phase_accept_solution(ready, &receiver, deposit, reward)
+	} verify {
+		assert_eq!(T::Currency::free_balance(&receiver), initial_balance + 20u32.into());
+		assert_eq!(T::Currency::reserved_balance(&receiver), 0u32.into());
+	}
+
+	finalize_signed_phase_reject_solution {
+		let receiver = account("receiver", 0, SEED);
+		let initial_balance = T::Currency::minimum_balance() * 10u32.into();
+		let deposit: BalanceOf<T> = 10u32.into();
+		T::Currency::make_free_balance_be(&receiver, initial_balance);
+		assert_ok!(T::Currency::reserve(&receiver, deposit));
+
+		assert_eq!(T::Currency::free_balance(&receiver), initial_balance - 10u32.into());
+		assert_eq!(T::Currency::reserved_balance(&receiver), 10u32.into());
+	}: {
+		<TwoPhase<T>>::finalize_signed_phase_reject_solution(&receiver, deposit)
+	} verify {
+		assert_eq!(T::Currency::free_balance(&receiver), initial_balance - 10u32.into());
+		assert_eq!(T::Currency::reserved_balance(&receiver), 0u32.into());
+	}
+
+	#[extra]
+	create_snapshot {
+		assert!(<TwoPhase<T>>::snapshot().is_none());
+	}: {
+		<TwoPhase::<T>>::create_snapshot()
+	} verify {
+		assert!(<TwoPhase<T>>::snapshot().is_some());
+	}
+
+	submit {
+		let c in 1 .. (T::MaxSignedSubmissions::get() - 1);
+
+		// the solution will be worse than all of them meaning the score need to be checked against all.
+		let solution = RawSolution { score: [(1000_0000u128 - 1).into(), 0, 0], ..Default::default() };
```

Comment:
```suggestion
		let solution = RawSolution { score: [(10_000_000u128 - 1).into(), 0, 0], ..Default::default() };
```

what kind of serial killer number separator is this!?!

Reply:
same below. I dunno how you separate with 4 zeros :)

Reply:
That is my eyesight getting weak.